bundler/lib/bundler/resolver/candidate.rb

```diff
@@ -48,35 +48,35 @@ def segments
         @version.segments
       end
 
-      def sort_obj
-        [@version, @priority]
-      end
-
       def <=>(other)
         return unless other.is_a?(self.class)
 
-        sort_obj <=> other.sort_obj
+        version <=> other.version || priority <=> other.priority
```

Comment:
<=> can return 0, so I think you might need a check for that

Reply:
Ah, right... `0 || 1 == 0`

Reply:
The idion is `(version <=> other.version).nonzero? or (priority <=> other.priority)`